### PR TITLE
Bugfix/basemodel handles non tensor raw predictions

### PR DESCRIPTION
**Pull request by JakeForsey**
DeepAR output contains a string value which can not be concatenated with torch.cat(). This MR resolves this by only concatenating values which are torch.Tensors

### CODE REVIEW

**Comment by jdb78** [comment]
We could instead use different functions for concatenations? Maybe np.concat for non-tensors?

**Comment by jdb78** [comment]
Great addition, btw!

**Comment by JakeForsey** [comment]
> We could instead use different functions for concatenations? Maybe np.concat for non-tensors?

Like this?

```
            for name in output[0].keys():
                v0 = output[0][name]
                if isinstance(v0, torch.Tensor):
                    output_cat[name] = torch.cat([out[name] for out in output], dim=0)
                elif isinstance(v0, np.ndarray):
                    output_cat[name] = np.concatenate([out[name] for out in output])
                else:
                    output_cat[name] = [out[name] for out in output]
```

Not sure if along which axis I should be concatenating numpy arrays

**Comment by jdb78** [comment]
Yes. This could work. Think by default along `axis=0`.

**Comment by JakeForsey** [comment]
> Yes. This could work. Think by default along `axis=0`.

isinstance(v0, np.ndarray) is not enough to ensure that np.concatenate works (zero-dimensional arrays and strings both through errors) so this is falling back to list comprehension.
